specification/app/resource-manager/Microsoft.App/preview/2023-05-01-preview/Builds.json

```diff
@@ -0,0 +1,529 @@
+        }
+      },
+      "description": "Configuration of the build."
+    },
+    "BuildProperties": {
+      "type": "object",
+      "properties": {
+        "provisioningState": {
+          "$ref": "#/definitions/ProvisioningState",
+          "description": "Build provisioning state.",
+          "readOnly": true
+        },
+        "buildStatus": {
+          "$ref": "#/definitions/BuildStatus",
+          "description": "Status of the build once it has been provisioned.",
+          "readOnly": true
+        },
+        "destinationContainerRegistry": {
+          "$ref": "#/definitions/ContainerRegistryWithCustomImage",
+          "description": "Container registry that the final image will be uploaded to."
+        },
+        "configuration": {
+          "$ref": "#/definitions/BuildConfiguration",
+          "description": "Configuration of the build."
+        },
+        "uploadEndpoint": {
+          "type": "string",
+          "description": "Endpoint to which the source code should be uploaded.",
+          "readOnly": true
+        }
+      },
+      "description": "The build properties."
+    },
+    "BuildResource": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/BuildProperties",
+          "description": "The resource-specific properties for this resource.",
+          "x-ms-client-flatten": true,
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        }
+      },
+      "description": "Information pertaining to an individual build.",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
+        }
+      ]
+    },
+    "BuildCollection": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/BuildResource"
+          },
+          "description": "The BuildResource items on this page"
+        },
+        "nextLink": {
+          "type": "string",
+          "format": "uri",
+          "description": "The link to the next page of items"
+        }
+      },
+      "description": "The response of a BuildResource list operation.",
+      "required": [
+        "value"
+      ]
+    },
+    "BuildStatus": {
+      "type": "string",
+      "description": "Status of the build once it has been provisioned.",
+      "enum": [
+        "NotStarted",
+        "InProgress",
+        "Succeeded",
+        "Cancelled",
+        "Failed"
+      ],
+      "x-ms-enum": {
+        "name": "BuildStatus",
+        "modelAsString": true
+      }
+    },
+    "ContainerRegistryWithCustomImage": {
+      "type": "object",
+      "properties": {
+        "server": {
+          "type": "string",
+          "description": "Login server of the container registry that the final image should be uploaded to."
+        },
+        "identity": {
```

Comment:
Is this a user assigned identity that is attached to the builder?

Reply:
Yes. This identity would need to be attached to the builder. Some validations would need to happen on our side to ensure that this is the case, and it should hard fail with a helpful error message if it isn't.